### ISSUE TITLE
Not able to delete assistants.

### ISSUE THREAD

**Issue by sangee2004** [rubra-ai/rubra#32]
**Steps to reproduce the problem:**

1. Created an assistant.
2. Deleted this assistant.
Assistant continues to be present in the `Explore_Assistants` page and is also listed as one of the available assistants in the `Chat` page and continues to be usable.

**Expected Behavior:**
Deleted this assistant should not be shown in  `Explore_Assistants` and `Chat` page

**Comment by rhm2k** [comment]
I mistakenly added too many files to an assistant. After deleting some, I later wanted to delete the assistant.
After 'running' for about 45 seconds, I receive the message:
Failed to delete: Internal Server Error

**Comment by sangee2004** [comment]
With v0.0.2 , I am seeing the assistant delete report errors and see the following exception in logs

```
2024-02-17 17:02:00 ERROR:app.vector_db.milvus.query_milvus:Failed to create new connection using: f5082f40e190421bb062d5e3c08c4935
2024-02-17 17:02:00 INFO:     172.18.0.10:43250 - "DELETE /assistants/asst_693acf HTTP/1.1" 500 Internal Server Error
2024-02-17 17:02:00 ERROR:    Exception in ASGI application
2024-02-17 17:02:00 Traceback (most recent call last):
2024-02-17 17:02:00   File "/usr/local/lib/python3.10/site-packages/pymilvus/client/grpc_handler.py", line 133, in _wait_for_channel_ready
2024-02-17 17:02:00     grpc.channel_ready_future(self._channel).result(timeout=timeout)
2024-02-17 17:02:00   File "/usr/local/lib/python3.10/site-packages/grpc/_utilities.py", line 162, in result
2024-02-17 17:02:00     self._block(timeout)
2024-02-17 17:02:00   File "/usr/local/lib/python3.10/site-packages/grpc/_utilities.py", line 106, in _block
2024-02-17 17:02:00     raise grpc.FutureTimeoutError()
2024-02-17 17:02:00 grpc.FutureTimeoutError
2024-02-17 17:02:00 
2024-02-17 17:02:00 The above exception was the direct cause of the following exception:
2024-02-17 17:02:00 
2024-02-17 17:02:00 Traceback (most recent call last):
2024-02-17 17:02:00   File "/usr/local/lib/python3.10/site-packages/uvicorn/protocols/http/h11_impl.py", line 408, in run_asgi
2024-02-17 17:02:00     result = await app(  # type: ignore[func-returns-value]
2024-02-17 17:02:00   File "/usr/local/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 84, in __call__
2024-02-17 17:02:00     return await self.app(scope, receive, send)
2024-02-17 17:02:00   File "/usr/local/lib/python3.10/site-packages/fastapi/applications.py", line 1106, in __call__
2024-02-17 17:02:00     await super().__call__(scope, receive, send)
2024-02-17 17:02:00   File "/usr/local/lib/python3.10/site-packages/starlette/applications.py", line 122, in __call__
2024-02-17 17:02:00     await self.middleware_stack(scope, receive, send)
2024-02-17 17:02:00   File "/usr/local/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
2024-02-17 17:02:00     raise exc
2024-02-17 17:02:00   File "/usr/local/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
2024-02-17 17:02:00     await self.app(scope, receive, _send)
2024-02-17 17:02:00   File "/usr/local/lib/python3.10/site-packages/starlette/middleware/cors.py", line 83, in __call__
2024-02-17 17:02:00     await self.app(scope, receive, send)
2024-02-17 17:02:00   File "/usr/local/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
2024-02-17 17:02:00     raise exc
2024-02-17 17:02:00   File "/usr/local/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
2024-02-17 17:02:00     await self.app(scope, receive, sender)
2024-02-17 17:02:00   File "/usr/local/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 20, in __call__
2024-02-17 17:02:00     raise e
2024-02-17 17:02:00   File "/usr/local/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 17, in __call__
2024-02-17 17:02:00     await self.app(scope, receive, send)
2024-02-17 17:02:00   File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 718, in __call__
2024-02-17 17:02:00     await route.handle(scope, receive, send)
2024-02-17 17:02:00   File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 276, in handle
2024-02-17 17:02:00     await self.app(scope, receive, send)
2024-02-17 17:02:00   File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 66, in app
2024-02-17 17:02:00     response = await func(request)
2024-02-17 17:02:00   File "/usr/local/lib/python3.10/site-packages/fastapi/routing.py", line 274, in app
2024-02-17 17:02:00     raw_response = await run_endpoint_function(
2024-02-17 17:02:00   File "/usr/local/lib/python3.10/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
2024-02-17 17:02:00     return await dependant.call(**values)
2024-02-17 17:02:00   File "/app/app/backend.py", line 401, in delete_assistant
2024-02-17 17:02:00     drop_collection(assistant_id)
2024-02-17 17:02:00   File "/app/app/vector_db/milvus/main.py", line 37, in drop_collection
2024-02-17 17:02:00     load_collection(collection_name).drop_collection()
2024-02-17 17:02:00   File "/app/app/vector_db/milvus/main.py", line 41, in load_collection
2024-02-17 17:02:00     return Milvus(
2024-02-17 17:02:00   File "/app/app/vector_db/milvus/query_milvus.py", line 134, in __init__
2024-02-17 17:02:00     self.alias = self._create_connection_alias(connection_args)
2024-02-17 17:02:00   File "/app/app/vector_db/milvus/query_milvus.py", line 208, in _create_connection_alias
2024-02-17 17:02:00     raise e
2024-02-17 17:02:00   File "/app/app/vector_db/milvus/query_milvus.py", line 203, in _create_connection_alias
2024-02-17 17:02:00     connections.connect(alias=alias, **connection_args)
2024-02-17 17:02:00   File "/usr/local/lib/python3.10/site-packages/pymilvus/orm/connections.py", line 356, in connect
2024-02-17 17:02:00     connect_milvus(**kwargs, user=user, password=password, token=token, db_name=db_name)
2024-02-17 17:02:00   File "/usr/local/lib/python3.10/site-packages/pymilvus/orm/connections.py", line 302, in connect_milvus
2024-02-17 17:02:00     gh._wait_for_channel_ready(timeout=timeout)
2024-02-17 17:02:00   File "/usr/local/lib/python3.10/site-packages/pymilvus/client/grpc_handler.py", line 136, in _wait_for_channel_ready
2024-02-17 17:02:00     raise MilvusException(
2024-02-17 17:02:00 pymilvus.exceptions.MilvusException: <MilvusException: (code=2, message=Fail connecting to server on localhost:19530. Timeout)>
2024-02-17 17:02:01 WARNING:root:assistant file file_5ebf34 for assistant asst_693acf not found
2024-02-17 17:02:11 ERROR:app.vector_db.milvus.query_milvus:Failed to create new connection using: 0d43c08ea669480f92ed19001cfd3cc8
2024-02-17 17:02:11 INFO:     172.18.0.10:55066 - "DELETE /assistants/asst_693acf HTTP/1.1" 500 Internal Server Error
2024-02-17 17:02:11 ERROR:    Exception in ASGI application
2024-02-17 17:02:11 Traceback (most recent call last):
2024-02-17 17:02:11   File "/usr/local/lib/python3.10/site-packages/pymilvus/client/grpc_handler.py", line 133, in _wait_for_channel_ready
2024-02-17 17:02:11     grpc.channel_ready_future(self._channel).result(timeout=timeout)
2024-02-17 17:02:11   File "/usr/local/lib/python3.10/site-packages/grpc/_utilities.py", line 162, in result
2024-02-17 17:02:11     self._block(timeout)
2024-02-17 17:02:11   File "/usr/local/lib/python3.10/site-packages/grpc/_utilities.py", line 106, in _block
2024-02-17 17:02:11     raise grpc.FutureTimeoutError()
2024-02-17 17:02:11 grpc.FutureTimeoutError
2024-02-17 17:02:11 
2024-02-17 17:02:11 The above exception was the direct cause of the following exception:
2024-02-17 17:02:11 
2024-02-17 17:02:11 Traceback (most recent call last):
2024-02-17 17:02:11   File "/usr/local/lib/python3.10/site-packages/uvicorn/protocols/http/h11_impl.py", line 408, in run_asgi
2024-02-17 17:02:11     result = await app(  # type: ignore[func-returns-value]
2024-02-17 17:02:11   File "/usr/local/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 84, in __call__
2024-02-17 17:02:11     return await self.app(scope, receive, send)
2024-02-17 17:02:11   File "/usr/local/lib/python3.10/site-packages/fastapi/applications.py", line 1106, in __call__
2024-02-17 17:02:11     await super().__call__(scope, receive, send)
2024-02-17 17:02:11   File "/usr/local/lib/python3.10/site-packages/starlette/applications.py", line 122, in __call__
2024-02-17 17:02:11     await self.middleware_stack(scope, receive, send)
2024-02-17 17:02:11   File "/usr/local/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
2024-02-17 17:02:11     raise exc
2024-02-17 17:02:11   File "/usr/local/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
2024-02-17 17:02:11     await self.app(scope, receive, _send)
2024-02-17 17:02:11   File "/usr/local/lib/python3.10/site-packages/starlette/middleware/cors.py", line 83, in __call__
2024-02-17 17:02:11     await self.app(scope, receive, send)
2024-02-17 17:02:11   File "/usr/local/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
2024-02-17 17:02:11     raise exc
2024-02-17 17:02:11   File "/usr/local/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
2024-02-17 17:02:11     await self.app(scope, receive, sender)
2024-02-17 17:02:11   File "/usr/local/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 20, in __call__
2024-02-17 17:02:11     raise e
2024-02-17 17:02:11   File "/usr/local/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 17, in __call__
2024-02-17 17:02:11     await self.app(scope, receive, send)
2024-02-17 17:02:11   File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 718, in __call__
2024-02-17 17:02:11     await route.handle(scope, receive, send)
2024-02-17 17:02:11   File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 276, in handle
2024-02-17 17:02:11     await self.app(scope, receive, send)
2024-02-17 17:02:11   File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 66, in app
2024-02-17 17:02:11     response = await func(request)
2024-02-17 17:02:11   File "/usr/local/lib/python3.10/site-packages/fastapi/routing.py", line 274, in app
2024-02-17 17:02:11     raw_response = await run_endpoint_function(
2024-02-17 17:02:11   File "/usr/local/lib/python3.10/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
2024-02-17 17:02:11     return await dependant.call(**values)
2024-02-17 17:02:11   File "/app/app/backend.py", line 401, in delete_assistant
2024-02-17 17:02:11     drop_collection(assistant_id)
2024-02-17 17:02:11   File "/app/app/vector_db/milvus/main.py", line 37, in drop_collection
2024-02-17 17:02:11     load_collection(collection_name).drop_collection()
2024-02-17 17:02:11   File "/app/app/vector_db/milvus/main.py", line 41, in load_collection
2024-02-17 17:02:11     return Milvus(
2024-02-17 17:02:11   File "/app/app/vector_db/milvus/query_milvus.py", line 134, in __init__
2024-02-17 17:02:11     self.alias = self._create_connection_alias(connection_args)
2024-02-17 17:02:11   File "/app/app/vector_db/milvus/query_milvus.py", line 208, in _create_connection_alias
2024-02-17 17:02:11     raise e
2024-02-17 17:02:11   File "/app/app/vector_db/milvus/query_milvus.py", line 203, in _create_connection_alias
2024-02-17 17:02:11     connections.connect(alias=alias, **connection_args)
2024-02-17 17:02:11   File "/usr/local/lib/python3.10/site-packages/pymilvus/orm/connections.py", line 356, in connect
2024-02-17 17:02:11     connect_milvus(**kwargs, user=user, password=password, token=token, db_name=db_name)
2024-02-17 17:02:11   File "/usr/local/lib/python3.10/site-packages/pymilvus/orm/connections.py", line 302, in connect_milvus
2024-02-17 17:02:11     gh._wait_for_channel_ready(timeout=timeout)
2024-02-17 17:02:11   File "/usr/local/lib/python3.10/site-packages/pymilvus/client/grpc_handler.py", line 136, in _wait_for_channel_ready
2024-02-17 17:02:11     raise MilvusException(
2024-02-17 17:02:11 pymilvus.exceptions.MilvusException: <MilvusException: (code=2, message=Fail connecting to server on localhost:19530. Timeout)>
2024-02-17 17:02:13 WARNING:root:assistant file file_5ebf34 for assistant asst_693acf not found
2024-02-17 17:02:23 ERROR:app.vector_db.milvus.query_milvus:Failed to create new connection using: 68970d0837cb40cabb44bbeda2f930be
2024-02-17 17:02:23 INFO:     172.18.0.10:44404 - "DELETE /assistants/asst_693acf HTTP/1.1" 500 Internal Server Error
2024-02-17 17:02:23 ERROR:    Exception in ASGI application
2024-02-17 17:02:23 Traceback (most recent call last):
2024-02-17 17:02:23   File "/usr/local/lib/python3.10/site-packages/pymilvus/client/grpc_handler.py", line 133, in _wait_for_channel_ready
2024-02-17 17:02:23     grpc.channel_ready_future(self._channel).result(timeout=timeout)
2024-02-17 17:02:23   File "/usr/local/lib/python3.10/site-packages/grpc/_utilities.py", line 162, in result
2024-02-17 17:02:23     self._block(timeout)
2024-02-17 17:02:23   File "/usr/local/lib/python3.10/site-packages/grpc/_utilities.py", line 106, in _block
2024-02-17 17:02:23     raise grpc.FutureTimeoutError()
2024-02-17 17:02:23 grpc.FutureTimeoutError
2024-02-17 17:02:23 
2024-02-17 17:02:23 The above exception was the direct cause of the following exception:
2024-02-17 17:02:23 
2024-02-17 17:02:23 Traceback (most recent call last):
2024-02-17 17:02:23   File "/usr/local/lib/python3.10/site-packages/uvicorn/protocols/http/h11_impl.py", line 408, in run_asgi
2024-02-17 17:02:23     result = await app(  # type: ignore[func-returns-value]
2024-02-17 17:02:23   File "/usr/local/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 84, in __call__
2024-02-17 17:02:23     return await self.app(scope, receive, send)
2024-02-17 17:02:23   File "/usr/local/lib/python3.10/site-packages/fastapi/applications.py", line 1106, in __call__
2024-02-17 17:02:23     await super().__call__(scope, receive, send)
2024-02-17 17:02:23   File "/usr/local/lib/python3.10/site-packages/starlette/applications.py", line 122, in __call__
2024-02-17 17:02:23     await self.middleware_stack(scope, receive, send)
2024-02-17 17:02:23   File "/usr/local/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
2024-02-17 17:02:23     raise exc
2024-02-17 17:02:23   File "/usr/local/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
2024-02-17 17:02:23     await self.app(scope, receive, _send)
2024-02-17 17:02:23   File "/usr/local/lib/python3.10/site-packages/starlette/middleware/cors.py", line 83, in __call__
2024-02-17 17:02:23     await self.app(scope, receive, send)
2024-02-17 17:02:23   File "/usr/local/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
2024-02-17 17:02:23     raise exc
2024-02-17 17:02:23   File "/usr/local/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
2024-02-17 17:02:23     await self.app(scope, receive, sender)
2024-02-17 17:02:23   File "/usr/local/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 20, in __call__
2024-02-17 17:02:23     raise e
2024-02-17 17:02:23   File "/usr/local/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 17, in __call__
2024-02-17 17:02:23     await self.app(scope, receive, send)
2024-02-17 17:02:23   File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 718, in __call__
2024-02-17 17:02:23     await route.handle(scope, receive, send)
2024-02-17 17:02:23   File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 276, in handle
2024-02-17 17:02:23     await self.app(scope, receive, send)
2024-02-17 17:02:23   File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 66, in app
2024-02-17 17:02:23     response = await func(request)
2024-02-17 17:02:23   File "/usr/local/lib/python3.10/site-packages/fastapi/routing.py", line 274, in app
2024-02-17 17:02:23     raw_response = await run_endpoint_function(
2024-02-17 17:02:23   File "/usr/local/lib/python3.10/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
2024-02-17 17:02:23     return await dependant.call(**values)
2024-02-17 17:02:23   File "/app/app/backend.py", line 401, in delete_assistant
2024-02-17 17:02:23     drop_collection(assistant_id)
2024-02-17 17:02:23   File "/app/app/vector_db/milvus/main.py", line 37, in drop_collection
2024-02-17 17:02:23     load_collection(collection_name).drop_collection()
2024-02-17 17:02:23   File "/app/app/vector_db/milvus/main.py", line 41, in load_collection
2024-02-17 17:02:23     return Milvus(
2024-02-17 17:02:23   File "/app/app/vector_db/milvus/query_milvus.py", line 134, in __init__
2024-02-17 17:02:23     self.alias = self._create_connection_alias(connection_args)
2024-02-17 17:02:23   File "/app/app/vector_db/milvus/query_milvus.py", line 208, in _create_connection_alias
2024-02-17 17:02:23     raise e
2024-02-17 17:02:23   File "/app/app/vector_db/milvus/query_milvus.py", line 203, in _create_connection_alias
2024-02-17 17:02:23     connections.connect(alias=alias, **connection_args)
2024-02-17 17:02:23   File "/usr/local/lib/python3.10/site-packages/pymilvus/orm/connections.py", line 356, in connect
2024-02-17 17:02:23     connect_milvus(**kwargs, user=user, password=password, token=token, db_name=db_name)
2024-02-17 17:02:23   File "/usr/local/lib/python3.10/site-packages/pymilvus/orm/connections.py", line 302, in connect_milvus
2024-02-17 17:02:23     gh._wait_for_channel_ready(timeout=timeout)
2024-02-17 17:02:23   File "/usr/local/lib/python3.10/site-packages/pymilvus/client/grpc_handler.py", line 136, in _wait_for_channel_ready
2024-02-17 17:02:23     raise MilvusException(
2024-02-17 17:02:23 pymilvus.exceptions.MilvusException: <MilvusException: (code=2, message=Fail connecting to server on localhost:19530. Timeout)>
```

**Comment by tybalex** [comment]
this is addressed in PR #89 , commit [4b83518](https://github.com/rubra-ai/rubra/pull/89/commits/4b835186dd6e17c8f6c2b261e588b27d1b24af3f)

**Comment by sangee2004** [comment]
With latest version of rubra , we are able to delete assistants (that were created with files) successfully.
